core/src/main/java/org/apache/spark/memory/TaskMemoryManager.java

```diff
@@ -389,12 +389,6 @@ public long cleanUpAllAllocatedMemory() {
       consumers.clear();
     }
 
-    for (MemoryBlock page : pageTable) {
```

Comment:
@davies in L382 we null out the `pageTable` so this loop doesn't do anything. However, I don't think we should delete this code. Should we just remove L382? Is this a cause for memory leaks?